src/ui/public/highlight/highlight_request.js

```diff
@@ -9,7 +9,7 @@ const FRAGMENT_SIZE = Math.pow(2, 31) - 1; // Max allowed value for fragment_siz
 function getHighlightQuery(query) {
   const clone = _.cloneDeep(query);
 
-  if (_.has(clone, 'query_string')) {
+  if (_.has(clone, 'query_string') && !_.has(clone, ['query_string', 'default_field'])) {
```

Comment:
No, the `_.has` array syntax is equivalent to saying `'query_string.default_field'`.